AK/FixedArray.h

```diff
@@ -36,6 +36,29 @@ class FixedArray {
         return MUST(try_create(size));
     }
 
+    // NOTE:
+    // Even though it may look like there will be a template instantiation of these functions for every size,
+    // the compiler will inline these anyway and therefore not generate any duplicate code.
+
+    template<size_t N>
+    static ErrorOr<FixedArray<T>> try_create(T(&&array)[N])
+    {
+        if (N == 0)
+            return FixedArray<T>();
+        T* elements = static_cast<T*>(kmalloc_array(N, sizeof(T)));
+        if (!elements)
+            return Error::from_errno(ENOMEM);
+        for (size_t i = 0; i < N; ++i)
+            new (&elements[i]) T(move(array[i]));
+        return FixedArray<T>(N, elements);
+    }
+
+    template<size_t N>
+    static FixedArray<T> must_create_but_fixme_should_propagate_errors(T(&&array)[N])
+    {
+        return MUST(try_create(move(array)));
+    }
```

Comment:
If you're adding a new API, why do you add this? If someone needs it for now, just `MUST()` the above call.

Reply:
That doesn't make it obvious that it's a FIXME. I'd be more happy to do so if we had `MUST_BUT_FIXME_SHOULD_PROPAGATE_ERRORS()`.

Reply:
By the way, the reason I think these APIs are useful is that it extremely lowers the barrier to convert a user to `FixedArray` from some other class like `Vector` that's not OOM-aware. If you create the latter in the constructor, you can simply use the `must_foo_but_fixme_should_propagate_errors()` to create the former. But if we force any new user to use error propagation, they will need to restructure a lot more code in order to make the switch.

Reply:
Sure, you convinced me, we'll leave the API for the time being.